pypika/clickhouse/dates_and_times.py

```diff
@@ -1,6 +1,101 @@
+import abc
+
+from pypika import CustomFunction
 from pypika.terms import Function
 
 
+ToYYYYMM = CustomFunction('toYYYYMM')
+
+
 class FormatDateTime(Function):
     def __init__(self, term, alt, **kwargs):
         super().__init__('formatDateTime', term, alt, **kwargs)
+
+
+class _AbstractAddSubtract(Function, metaclass=abc.ABCMeta):
+    def __init__(self, name, interval: int, *args, **kwargs):
+        super().__init__(self._get_clickhouse_name(), name, interval, *args, **kwargs)
+
+    @abc.abstractmethod
+    def _get_clickhouse_name(self) -> str:
+        pass
+
+
+class AddYears(_AbstractAddSubtract):
```

Comment:
These could all be simplified to use `CustomFunction`, like

```
_add_subtract_args = ['name', 'interval']
AddYears = CustomFunction('addYears', _add_subtract_args)
AddMonths = CustomFunction('addMonths', _add_subtract_args)
...
```

That would make them all one liners

Reply:
Sorry for delay. Crazy week. 
Fixed.